packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -124,6 +124,35 @@ void main() {
         ),
       ),
     );
+    await tester.pumpAndSettle();
+
+    // Overall height for this InputDecorator is 56dps:
+    //   12 - top padding
+    //   12 - floating label (ahem font size 16dps * 0.75 = 12)
+    //    4 - floating label / input text gap
+    //   16 - input text (ahem font size 16dps)
+    //   12 - bottom padding
+
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 56.0));
+    expect(tester.getTopLeft(find.text('text')).dy, 28.0);
+    expect(tester.getBottomLeft(find.text('text')).dy, 44.0);
+    expect(tester.getTopLeft(find.text('label')).dy, 12.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 24.0);
+    expect(getBorderBottom(tester), 56.0);
+    expect(getBorderWeight(tester), 1.0);
+
+    // The label appears above the input text when empty and alwaysFloatLabel
```

Comment:
It doesn't look like there's anything verifying that the label is in a different position.  Maybe you could save the `find.text('label').dy` when alwaysFloatLabel is false and then pumpWidget again with alwaysFloatLabel true.  Then compare the default dy with the floated dy and expect the floated value to be `isGreaterThan` the default value.